v2/navigation.jsx

```diff
@@ -212,23 +223,33 @@ export function Navigation(props) {
         if (state.state !== 'settled') {
             return
         }
-        const url = new URL(window.location.href)
-        url.searchParams.delete('stack')
-        for (let string of state.stack) {
-            url.searchParams.append('stack', string)
+
+        if (state.via === 'push') {
+            const url = new URL(window.location.href)
+            url.searchParams.delete('stack')
+            for (let string of state.stack) {
+                url.searchParams.append('stack', string)
+            }
+            for (let [key, value] of Object.entries(state.params)) {
+                url.searchParams.set(key, value)
+            }
+            window.history.pushState({}, '', url)
         }
-        for (let [key, value] of Object.entries(state.params)) {
-            url.searchParams.set(key, value)
+
+        if (state.via === 'pop') {
+            window.history.go(-1)
         }
```

Comment:
This is the main change. Before I was just issuing a 'history.pushState' for forward AND back navigations. in iOS this didn't cause any issues, because the dashboard is always closed from a button within the dashboard - but on android the 'back' button on the device can also close the dashboard, and for that to occur the history stack has be empty.